### PR TITLE
make `flow` and `task` decorators pass strict typing

### PR DESCRIPTION
**Pull request by zzstoatzz**
aw man, that type completeness check is a tough cookie

### CODE REVIEW

File: src/prefect/flows.py [hunk 1]
```diff
@@ -1361,7 +1361,7 @@ def __call__(
         )
 
     @sync_compatible
-    async def visualize(self, *args, **kwargs):
+    async def visualize(self, *args: Any, **kwargs: Any):
```
Comment (desertaxle) [inline src/prefect/flows.py:1364]: We can do this right?
```suggestion
    async def visualize(self, *args: "P.args", **kwargs: "P.kwargs"):
```

File: src/prefect/flows.py [hunk 2]
```diff
@@ -1458,14 +1458,14 @@ def flow(
...
-    __fn=None,
+    __fn: Optional[Callable[P, R]] = None,
     *,
     name: Optional[str] = None,
     version: Optional[str] = None,
     flow_run_name: Optional[Union[Callable[[], str], str]] = None,
     retries: Optional[int] = None,
     retry_delay_seconds: Union[int, float, None] = None,
-    task_runner: Optional[TaskRunner] = None,
+    task_runner: Optional[TaskRunner[PrefectFuture[Any]]] = None,
```
Comment (desertaxle) [inline src/prefect/flows.py:1468]: Does this work?
```suggestion
    task_runner: Optional[TaskRunner[PrefectFuture[R]]] = None,
```

**Review by desertaxle — APPROVED** [review]
I think there are a couple of places that we could type stuff a little more specifically, but LGTM otherwise!

**Comment by zzstoatzz** [comment]
good calls! updated in [022f3bd](https://github.com/PrefectHQ/prefect/pull/16251/commits/022f3bdcc99c5afd5bb623699c3e8175eeb9f86d) (+ more!)

**Pull request merged**